feat: 채용공고 조회 기능 추가

k-syou opened the pull request:
## ✨ 개요
<!-- 변경/추가된 내용을 요약해주세요 -->
채용공고 조회 기능 추가
등록일 순 전체 조회 및 마감일 순 전체 조회 기능 추가
---

## ✅ 작업 내용
- [x] 기능 추가
- [ ] 버그 수정
- [ ] 코드 리팩토링
- [ ] 테스트 코드 작성
- [ ] 문서 수정

---

## 🧪 테스트 내용
<!-- 어떤 테스트를 했는지 간략하게 작성해주세요 -->
아직 테스트 못해봤습니다.
---

## 📎 관련 이슈
<!-- ex: Closes #23, Fixes #45 -->

---

## 📝 기타 참고 사항
<!-- 리뷰어가 참고해야 할 내용이 있다면 작성해주세요 -->

kimbro97 committed:
Merge pull request #1 from seoul-data-job-fit/develop

PR 템플릿 등록

k-syou committed:
feat: 채용공고 조회 기능 구현

- 채용공고 Model 생성
- 채용공고 레포지토리 생성
- 채용공고 컨트롤러 및 서비스 개발

kimbro97 commented on src/main/java/com/jobfit/server/domain/recruit/Recruit.java:
```diff
@@ -0,0 +1,161 @@
+  private String educationType;
+
+  @Column(name = "job_type")
+  private String jobType;
```
JPA entity를 사용하면 컬럼 지정해줄때 카멜케이스를 사용하면 일반적으로 스네이크케이스로(snake_case) 맵핑합니다
지금 코드를 보면 @Column은 필요없을거 같습니다!

k-syou replied:
넵 확인 감사합니다.

kimbro97 commented on src/main/java/com/jobfit/server/interfaces/api/recruit/RecruitController.java:
```diff
@@ -0,0 +1,28 @@
+
+	private final RecruitService recruitService;
+	@GetMapping("/api/v1/recruit/recent")
+	public List<RecruitInfo> getRecentRecruits() {
```
controller에서 응답해줄때는 ApiResponse를 사용해주세요!
둘 다 조회 API니까 ApiResponse.OK가 적당할거같습니다

k-syou replied:
넵 확인 했습니다

kimbro97 commented on src/main/java/com/jobfit/server/domain/recruit/Recruit.java:
```diff
@@ -0,0 +1,161 @@
+
+  @Id
+  @GeneratedValue(strategy = IDENTITY)
+  private Long id;
```
앞으로 프라이머리 키로 사용하는 id값은 recruit_id로 가져가면 좋을거같습니다
``` java
@Id
@Column(name = "recruit_id")
@GeneratedValue(strategy = IDENTITY)
private Long id

```

kimbro97 reviewed (commented):
리뷰 남겨드립니다!

k-syou committed:
refact: recruit 어노테이션 제거 및 컨트롤러 수정

- JPA 에서 자동으로 컬럼 이름 스네이크 케이스로 적용해주므로 Column 어노 테이션 제거
- 컨트롤러에서 return 객체 ApiResponse 로 변경

k-syou committed:
feat: 채용공고 상세조회 api 개발

kimjongsoo97 commented:
고생하셧습니다
